lib/node_modules/@stdlib/math/base/special/sincos/benchmark/julia/benchmark.jl

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env julia
+
+import BenchmarkTools
+
+# Benchmark variables:
+name = "sincos";
+repeats = 3;
+
+"""
+	print_version()
+
+Prints the TAP version.
+
+# Examples
+
+``` julia
+julia> print_version()
+```
+"""
+function print_version()
+	@printf( "TAP version 13\n" );
+end
+
+"""
+	print_summary( total, passing )
+
+Print the benchmark summary.
+
+# Arguments
+
+* `total`: total number of tests
+* `passing`: number of passing tests
+
+# Examples
+
+``` julia
+julia> print_summary( 3, 3 )
+```
+"""
+function print_summary( total, passing )
+	@printf( "#\n" );
+	@printf( "1..%d\n", total ); # TAP plan
+	@printf( "# total %d\n", total );
+	@printf( "# pass  %d\n", passing );
+	@printf( "#\n" );
+	@printf( "# ok\n" );
+end
+
+"""
+	print_results( iterations, elapsed )
+
+Print benchmark results.
+
+# Arguments
+
+* `iterations`: number of iterations
+* `elapsed`: elapsed time (in seconds)
+
+# Examples
+
+``` julia
+julia> print_results( 1000000, 0.131009101868 )
+```
+"""
+function print_results( iterations, elapsed )
+	rate = iterations / elapsed
+
+	@printf( "  ---\n" );
+	@printf( "  iterations: %d\n", iterations );
+	@printf( "  elapsed: %0.9f\n", elapsed );
+	@printf( "  rate: %0.9f\n", rate );
+	@printf( "  ...\n" );
+end
+
+"""
+	sincos( x )
+
+Compute the sine and cosine of a number (since sincos is not directly exposed in julia).
+
+# Arguments
+
+* `x`: argument, in radians
+
+# Examples
+
+``` julia
+julia> sincos( 0.0 )
+```
+"""
+function sincos( x )
+	[ sin( x ), cos( x ) ];
+end
+
+
+"""
+	benchmark()
+
+Run a benchmark.
+
+# Notes
+
+* Benchmark results are returned as a two-element array: [ iterations, elapsed ].
+* The number of iterations is not the true number of iterations. Instead, an 'iteration' is defined as a 'sample', which is a computed estimate for a single evaluation.
+* The elapsed time is in seconds.
+
+# Examples
+
+``` julia
+julia> benchmark();
+```
+"""
+function benchmark()
+	t = BenchmarkTools.@benchmark sincos( (20.0*rand()) - 10.0 ) samples=1e6
```

Comment:
Did you try running this? I believe this will not work since `sincos` is defined outside the macro. Prefix` sincos` with a `$`.